client/driver/lxc.go

```diff
@@ -0,0 +1,486 @@
+func (d *LxcDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
+	version := lxc.Version()
+	if version == "" {
+		return false, nil
+	}
+	node.Attributes["driver.lxc.version"] = version
+	node.Attributes["driver.lxc"] = "1"
+	return true, nil
+}
+
+// Start starts the LXC Driver
+func (d *LxcDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
+	var driverConfig LxcDriverConfig
+	if err := mapstructure.WeakDecode(task.Config, &driverConfig); err != nil {
+		return nil, err
+	}
+	lxcPath := lxc.DefaultConfigPath()
+	if path := d.config.Read("lxc.path"); path != "" {
+		lxcPath = path
+	}
+
+	containerName := fmt.Sprintf("%s-%s", task.Name, ctx.AllocID)
+	c, err := lxc.NewContainer(containerName, lxcPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create container: %v", err)
+	}
+
+	var verbosity lxc.Verbosity
+	switch driverConfig.Verbosity {
+	case "verbose":
+		verbosity = lxc.Verbose
+	case "", "quiet":
+		verbosity = lxc.Quiet
+	default:
+		return nil, fmt.Errorf("lxc driver config 'verbosity' can only be either quiet or verbose")
+	}
+	c.SetVerbosity(verbosity)
+
+	var logLevel lxc.LogLevel
+	switch driverConfig.LogLevel {
+	case "trace":
+		logLevel = lxc.TRACE
+	case "debug":
+		logLevel = lxc.DEBUG
+	case "info":
+		logLevel = lxc.INFO
+	case "warn":
+		logLevel = lxc.WARN
+	case "", "error":
+		logLevel = lxc.ERROR
+	default:
+		return nil, fmt.Errorf("lxc driver config 'log_level' can only be trace, debug, info, warn or error")
+	}
+	c.SetLogLevel(logLevel)
+
+	logFile := filepath.Join(ctx.AllocDir.LogDir(), fmt.Sprintf("%v-lxc.log", task.Name))
+	c.SetLogFile(logFile)
+
+	options := lxc.TemplateOptions{
+		Template:             driverConfig.Template,
+		Distro:               driverConfig.Distro,
+		Release:              driverConfig.Release,
+		Arch:                 driverConfig.Arch,
+		FlushCache:           driverConfig.FlushCache,
+		DisableGPGValidation: driverConfig.DisableGPGValidation,
+	}
+
+	if err := c.Create(options); err != nil {
+		return nil, fmt.Errorf("unable to create container: %v", err)
+	}
+
+	// Set the network type to none
+	if err := c.SetConfigItem("lxc.network.type", "none"); err != nil {
+		return nil, fmt.Errorf("error setting network type configuration: %v", err)
+	}
+
+	// Bind mount the shared alloc dir and task local dir in the container
+	taskLocalDir, ok := ctx.AllocDir.TaskDirs[task.Name]
+	if !ok {
+		return nil, fmt.Errorf("failed to find task local directory: %v", task.Name)
+	}
+	taskDirMount := fmt.Sprintf("%s local none rw,bind,create=dir", taskLocalDir)
+	allocDirMount := fmt.Sprintf("%s alloc none rw,bind,create=dir", ctx.AllocDir.SharedDir)
+
+	if err := c.SetConfigItem("lxc.mount.entry", allocDirMount); err != nil {
+		return nil, fmt.Errorf("error setting alloc dir bind mounts configuration: %v", err)
+	}
+	if err := c.SetConfigItem("lxc.mount.entry", taskDirMount); err != nil {
```

Comment:
Calling `SetConfigItem("lxc.mount.entry",...` twice will append the second entry, not overwrite the first?

I couldn't tell from looking at the lxc docs.